azb2c-with-parcel.md

```diff
@@ -0,0 +1,106 @@
+---
+title: Build Azure AD B2C Templates
+subtitle: A simplistic approach using parcel.js
+domain: campzulu.hashnode.dev
+tags: web-development, azure, build-tool, sass
+cover: https://cdn.hashnode.com/res/hashnode/image/upload/v1663572757122/uN-z_ruLg.jpg?auto=compress
+publishAs: csalv
+ignorePost: true
+---
+
+
+
+# The goal
+When integrating Azure Active Directory B2C service into your app, you can customize the [HTML with templates](https://docs.microsoft.com/en-us/azure/active-directory-b2c/customize-ui-with-html) to provide a more immersive user experience. This article will describe a simple approach on how to setup a local project to develop such templates and create a production ready build.
+
+The basics to do so are simple enough: Publish a public available HTML page with styles/js to your liking that includes a ```div``` with id ```api``` can be used by the Azure B2C engine to render its login input fields inside:
+
+```html
+<div id="api"></div>
+```
+
+But immediately some challenges came to my mind: How do we get a production ready optimized build (e.g. css transpiling, minifying, ...) as well as some improved developer experience (scss preprocessor)
+
+# The idea
+
+This question leads us to [parcel.js](https://parceljs.org/), as I need a simple build for the static web content. I didn't want to pick a JS Framework to reach that goal (but be able to add any lightweight JS content if necessary at a later stage) to just create a simple static HTML file with CSS.
```

Comment:
I still think this is a bit too steep in the topic. I maybe would start with something like this:

> Parcel is a zero-configuration build tool for the web. This seems to be a good fit for my requirements as I didn't want to pick a JS Framework to just create a simple static HTML file with CSS and JavaScript.